frontend/src/components/views.js

```diff
@@ -369,4 +419,242 @@ class CustomSQLQuery extends React.Component {
+        "IND-CPA": "Indistinguisability under an adaptive Chosen Plaintext Attack",
+        "EUF-CMA": "Existential Unforgeability under a Chosen Message Attack",
+        "SUF-CMA": "Strong Existential Unforgeability under a Chosen Message Attack",
+    }
+
+    constructor(props) {
+        super(props);
+        this.db = props.db;
+        this.params = qs.parse(this.props.history.location.search);
+
+        // TODO: use path not query string
+        var path = this.params['_'] ?? '';
+
+        this.state = {
+            path: path
+        }
+    }
+
+    renderOverview () {
+        var stmt = "SELECT id, name FROM scheme WHERE type=? ORDER BY name;";
+
+        return (
+            <Grid container justify="center" spacing={2} maxWidth="md">
+                {Object.keys(this.types).map(type => (
+                <Grid item>
+                    <Paper>
+                        <Box p={4}>
+                            <Typography variant="h6">{ this.types[type].name }s</Typography>
+                            <List>
+                                {queryAll(this.db, stmt, [type]).map(s => (
+                                    <ListItem key={s.id}><ListItemText>
+                                        <Link href={"?_=" + type + "/" + s.id}>{s.name}</Link>
+                                    </ListItemText></ListItem>
+                                ))}
+                            </List>
+                        </Box>
+                    </Paper>
+                </Grid>
+                ))}
+            </Grid>
+        );
+    }
+
+    renderScheme (type, id) {
+        let db = this.db; // `this` is overriden inside `map`s and the like
+
+        const s = queryAll(this.db, "SELECT * FROM scheme WHERE type=? AND id=?", [type, id])[0];
+        console.log('response', s);
+        if ( s === undefined ) {
+            return <Container><Paper>No such scheme.</Paper></Container>;
+        }
+
+        const authors = queryAll(db, "SELECT * FROM scheme_author WHERE scheme_id=?", [id]);
+        const problems = queryAll(db, "SELECT * FROM scheme_problem WHERE scheme_id=?", [id]);
+        const links = queryAll(db, "SELECT * FROM scheme_link WHERE scheme_id=?", [id]);
+        const sources = queryAll(db, "SELECT * FROM scheme_source WHERE scheme_id=?", [id]);
+        const flavors = queryAll(db, "SELECT * FROM flavor WHERE scheme_id=?", [id]);
+
+        let TypeIcon = this.types[s.type].icon;
+        return [
+            <Container maxWidth="md">
+            <Paper>
+            <Box>
+                <List>
+                    <ListItem key="head" alignItems="flex-start">
+                        <ListItemIcon>
+                            <Typography component="span" variant="h2">
+                                <Tooltip title={ this.types[s.type].name } arrow>
+                                    { <TypeIcon fontSize="large" titleAccess={ this.types[s.type].name } /> }
```

Comment:
Setting the `titleAccess` fields results in a double ToolTip for me.

Reply:
Thanks, fixed.